### PR TITLE
Add support for replacement variables in the Snippet Window.

### PR DESCRIPTION
**Pull request by herregroen**
Allows usage of fields added by ACF in the Snippet Window.

The fields can be used with %%cf_{name-of-custom-field}%%

This fixes Yoast/wordpress-seo#5632.

This depends on Yoast/wordpress-seo#7607 as it uses the exposed `ReplaceVar` class but will fail gracefully ( not do anything ) if it's not present.

### CODE REVIEW

File: js/src/replacevars.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    fieldData   = collect.getFieldData();
+    replaceVars = {}
+
+    _.each(fieldData, function(field) {
+        // Remove HTML tags using jQuery in case of a wysiwyg field.
+        var content = (field.type === 'wysiwyg') ? jQuery( jQuery.parseHTML( field.content) ).text() : field.content;
+
+        replaceVars[field.name] = new ReplaceVar( '%%cf_'+field.name+'%%', content, { source: 'direct' } );
+        YoastSEO.wp.replaceVarsPlugin.addReplacement( replaceVars[field.name] );
+        console.log("Created ReplaceVar for: ", field.name, " with: ", content, replaceVars[field.name]);
```
Comment (moorscode) [inline js/src/replacevars.js]: This should be wrapped with a `config.debug`

File: js/src/replacevars.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return;
+    }
+
+    fieldData   = collect.getFieldData();
+    _.each(fieldData, function(field) {
+        // Remove HTML tags using jQuery in case of a wysiwyg field.
+        var content = (field.type === 'wysiwyg') ? jQuery(jQuery.parseHTML(field.content)).text() : field.content;
+
+        replaceVars[field.name].replacement = content;
+        console.log("Updated ReplaceVar for: ", field.name, " with: ", content, replaceVars[field.name]);
```
Comment (moorscode) [inline js/src/replacevars.js]: This should be wrapped with a `config.debug`

File: js/src/replacevars.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    });
+};
+
+module.exports = {
+    createReplaceVars: createReplaceVars,
+    updateReplaceVars: updateReplaceVars
+};
```
Comment (moorscode) [inline js/src/replacevars.js]: Missing newline

File: js/src/collect/collect.js [hunk 4]
```diff
@@ -22,6 +17,17 @@ Collect.prototype.append = function(data){
         field_data = scraper_store.getScraper(type).scrape(field_data);
     });
 
+    return field_data
```
Comment (moorscode) [inline js/src/collect/collect.js]: Missing `;`

File: js/src/replacevars.js (hunk 3 above)
Comment (herregroen) [inline js/src/replacevars.js]: Has been fixed.

**Comment by kraftner** [comment]
Having a test for this would be nice.

**Comment by herregroen** [comment]
Tests have been written. Bugs discovered in the test have been fixed. Also changed two existing tests that were failing.

* The current class is not on the `#menu-dashboard` but a child.
* The image modal opens on the upload tab. Added code to switch to the Media Library.

**Comment by moorscode** [comment]
Acceptance done!

**Pull request merged**